daemon/builder/dockerfile/internals.go

```diff
@@ -279,46 +281,21 @@ func withoutHealthcheck() runConfigModifier {
 
 func copyRunConfig(runConfig *container.Config, modifiers ...runConfigModifier) *container.Config {
 	cfgCopy := *runConfig
-	cfgCopy.Cmd = copyStringSlice(runConfig.Cmd)
-	cfgCopy.Env = copyStringSlice(runConfig.Env)
-	cfgCopy.Entrypoint = copyStringSlice(runConfig.Entrypoint)
-	cfgCopy.OnBuild = copyStringSlice(runConfig.OnBuild)
-	cfgCopy.Shell = copyStringSlice(runConfig.Shell)
-
-	if cfgCopy.Volumes != nil {
-		cfgCopy.Volumes = make(map[string]struct{}, len(runConfig.Volumes))
-		for k, v := range runConfig.Volumes {
-			cfgCopy.Volumes[k] = v
-		}
-	}
-
-	if cfgCopy.ExposedPorts != nil {
-		cfgCopy.ExposedPorts = make(container.PortSet, len(runConfig.ExposedPorts))
-		for k, v := range runConfig.ExposedPorts {
-			cfgCopy.ExposedPorts[k] = v
-		}
-	}
-
-	if cfgCopy.Labels != nil {
```

Comment:
For those wondering; `nil` check shouldn't be needed because both `slices.Clone` and `maps.Clone` check for `nil` and returns it as-is.